lecture2/task2.c

```diff
@@ -0,0 +1,104 @@
+#include <stdio.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+typedef struct my_ll {
+	int number;
+	char name[16];
+	struct my_ll *next;
+} myll;
+
+int copy_digits(char *inputs,char *outputs)
+{
+	uint8_t i=0,j=0;
+	while(inputs[i])
+	{
+		if((inputs[i] >= '0')&&(inputs[i] <= '9'))
+		{
+			outputs[j++] = inputs[i];
+		}
+		i++;
+	}
+	outputs[j] = 0;
+	return 0;
+}
+
+int biggest(int *inputa, int size)
+{
+	int bige = inputa[0];
+	for(int i = 1;i<size;i++)
+	{
+		if(bige < inputa[i]) bige = inputa[i];
+	}
+	return bige;
+}
+
+myll *ListAdd(myll *pnext)
+{
+	myll *nem = (myll *)malloc(sizeof(myll));
+	printf("input name (max 15 charcters):\n");
+	scanf("%s",nem->name);
+	printf("input number:\n");
+	scanf("%d",&nem->number);
+	nem->next = pnext;
+	return nem;
+}
+
+int *ListLs(myll *pnext)
+{
+	myll *pn = pnext;
+	while(pn != NULL)
+	{
+		printf("%s   %d\n",pn->name,pn->number);
+		pn = pn->next;
+	}
+	return 0;
+}
+
+int *ListRemove(myll *pn)
```

Comment:
According to homework, ListRemove function was supposed to remove a single element from the linked list (if found). With your implementation, it removes the whole list.

Consider adding a new function ListRemoveElement or changing this function to remove just a single element.